roles/satellite-clone/files/disassociate_capsules.rb

```diff
@@ -24,8 +24,11 @@ def capsule_lce_args(action, capsule_id, env)
   "--csv capsule content #{action}-lifecycle-environment --id #{capsule_id} --lifecycle-environment-id #{env}"
 end
 
+# the correct way is to look at hosts with the Pulpcore feature and the mirror capability
+# but hammer doesn't support that
+internal_capsule = ARGV[0]
 external_capsules = []
-external_capsule_ids = get_info_from_hammer("--csv capsule list --search 'feature = \"Pulp Node\"'")
+external_capsule_ids = get_info_from_hammer("--csv capsule list --search '(feature = \"Pulp Node\" or feature = \"Pulpcore\") and name != \"#{internal_capsule}\"'")
```

Comment:
```suggestion
external_capsule_ids = get_info_from_hammer("--csv capsule list --search '(name != \"#{internal_capsule}\"'")
```

I think this is simpler and works just as well; downstream does not support disabling Pulp on Capsules

Reply:
That's probably true.

One thing that bothers me: you can update the name of a capsule, and nothing will break… or should, with the code change here, it will :D

Reply:
We could do `url !~ https://#{internal_capsule}` instead, but that also doesn't feel right

Reply:
I would prefer not to build that in. The whole concept of a Smart Proxy on the same host is IMHO a Katelloism I'd love to kill. As we're looking at other architectures, we may at some point support a `satellite.example.com` without a Capsule and then a `content.example.com` which is your primary Capsule. If you start relying only on hostnames it becomes even more obscure that you really do want Smart Proxies that have content.

> I think this is simpler and works just as well; downstream does not support disabling Pulp on Capsules

It's not supported today, but your suggestion makes it harder to support it tomorrow.

We should really file an RFE for Hammer to expose the capability. The API does.